### PR TITLE
fix types: update validatorCompiler options

### PR DESCRIPTION
**Pull request by joshuaavalon**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Related issues: #3457

### CODE REVIEW

**Comment by Eomm** [comment]
Would you mind to add a test [here](https://github.com/fastify/fastify/blob/master/test/types/index.ts)?

**Comment by joshuaavalon** [comment]
> Would you mind to add a test [here](https://github.com/fastify/fastify/blob/master/test/types/index.ts)?

Where are you referring to? I got 404 with your link, Do you mean add a test to `test/types`?

**Comment by Eomm** [comment]
Yeah, fixed the link https://github.com/fastify/fastify/tree/main/test/types

File: fastify.d.ts [hunk 1]
```diff
@@ -137,15 +149,12 @@ export type FastifyServerOptions<
...
+      buildValidator?: (externalSchemas: Record<string, unknown>, options?: Ajv) => FastifySchemaCompiler<unknown>;
       buildSerializer?: (externalSchemas: unknown, serializerOptsServerOption: FastifyServerOptions["serializerOpts"]) => FastifySerializerCompiler<unknown>;
     };
   };
   return503OnClosing?: boolean,
-  ajv?: {
-    customOptions?: AjvOptions,
-    plugins?: Function[]
-  },
+  ajv?: Ajv,
```
Comment (climba03003) [inline fastify.d.ts:157]: Let's do the same as `serializerOpts`.

File: fastify.d.ts [hunk 2]
```diff
@@ -30,49 +29,56 @@ declare function fastify<
   Server extends http2.Http2SecureServer,
   Request extends RawRequestDefaultExpression<Server> = RawRequestDefaultExpression<Server>,
   Reply extends RawReplyDefaultExpression<Server> = RawReplyDefaultExpression<Server>,
-  Logger extends FastifyLoggerInstance = FastifyLoggerInstance
->(opts: FastifyHttp2SecureOptions<Server, Logger>): FastifyInstance<Server, Request, Reply, Logger> & PromiseLike<FastifyInstance<Server, Request, Reply, Logger>>
+  Logger extends FastifyLoggerInstance = FastifyLoggerInstance,
+  Ajv = FastifyAjvOptions
```
Comment (climba03003) [inline fastify.d.ts:33]: I do not think it is good to add one more parameter here.

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Comment by joshuaavalon** [comment]
I removed generics but I kept `FastifyAjvOptions`. By removing generics, users cannot pass the custom options, but keeping `FastifyAjvOptions` at least allows users to extends the interface in their code.

**Comment by mcollina** [comment]
@fastify/typescript could you take a look?

**Pull request closed without merging**